File: src/main/java/org/books/services/OwnerService.java
package org.books.services;

import org.books.beans.LoggedOwner;
import org.books.entities.Owner;
import org.books.persistence.OwnerRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class OwnerService {

    @Autowired
    private OwnerRepository repository;

    @Autowired
    private SecurityService securityService;

    public List<Owner> procurarOwner(Owner owner) {
        return repository.findByName(owner.getName());
    }

    public LoggedOwner criarOwner(Owner owner) {
        LoggedOwner loggedOwner = new LoggedOwner();

        loggedOwner.setName(owner.getName());
        loggedOwner.setToken(securityService.gerarToken(owner));

        return loggedOwner;
    }
}

File: src/main/java/org/books/services/OwnerService.kt
package org.books.services

import org.apache.commons.lang3.RandomStringUtils
import org.books.beans.LoggedOwner
import org.books.entities.Owner
import org.books.persistence.OwnerRepository
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class OwnerService {

    @Autowired
    private val repository: OwnerRepository? = null

    @Autowired
    private val securityService: SecurityService? = null

    fun procurarOwner(owner: Owner): List<Owner> {
        return repository!!.findByName(owner.name)
    }

    fun criarOwner(owner: Owner): LoggedOwner {
        val loggedOwner = LoggedOwner()

        owner.salt = RandomStringUtils.randomAlphanumeric(KEY_SIZE)
        owner.password = securityService!!.encryptPassword(owner.password, owner.salt)

        repository!!.save(owner)

        loggedOwner.id = owner.id
        loggedOwner.name = owner.name
        loggedOwner.token = securityService!!.gerarToken(owner)

        return loggedOwner
    }

    companion object {

        private val KEY_SIZE : Int = 16

    }
}
